R/mail.R

```diff
@@ -0,0 +1,62 @@
+# Kaiaulu - https://github.com/sailuh/kaiaulu
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+
+
+
+## Creates fake mbox data
+#' @param mlist specific mailing list associated with this mbox data
+#' @param sender information of person sending the mail
+#' @param date First year in the range to be downloaded
+#' @param subject subject of the email
+#' @param content the body of the email
+#' @param randomise_email_case If TRUE, randomizes the email case. (Default = TRUE)
+#' @export
+create_fake_mbox <- function(mlist, sender, date, subject, content, randomise_email_case = TRUE) {
```

Comment:
@Rubegen there is something I am finding weird here. Where is the "To" field on the e-mail?  

Also, I used parse_mbox() on an actual data file and the parser returns these columns:

[1] "reply_id"         "in_reply_to_id"   "reply_datetimetz" "reply_from"       "reply_to"        
[6] "reply_cc"         "reply_subject"    "reply_body" 

reply_cc was empty on my data, but you should try to find a case with Cc (you can try to google a few mod_mbox apache mailing lists to see if you find a case). I am not sure if the reply_ids are getting generated by Perceval, if it is then you should not add to the raw data, but I am curious what your fake data gets assigned for these columns after parse_mbox is called on it. 

In addition, some of these parameters need to be facilitated to the person calling them. For example, it is not easy to manually type the dates in the raw data format of mbox. So you want to make sure your input from the user calling the function is simpler (02/03/2023 15:30:01 HST) and the function does the rest to match mbox. Having to guess the abbreviation of the date (Aug, Mon, etc) will make using this function a tedious and erroneous process for any user for example, as they would need to check the documentation. Make sure to include an example on @param too of the format expected.

The same is true to sender. The sender format is `"Firstname Lastname" <email@domain.org>` (note the inclusion of double quotes). It is easier if your function signature ask a sender_name parameter, and an sender_email parameter, that then insert in the mbox with the caveats, than telling in the docs for the person calling the function to not forget double quotes, one white space and <> symbols. 

You should also rename these parameters to match the columns Kaiaulu already uses. Afterall, if we are using this to unit test, it is easier to manipulate the data if the naming of the columns are consistent. So sender should be `reply_from` etc. This only doesn't apply if for some reason the parser function has additional columns or it is easier to break them apart (so you would have here reply_from_name and reply_from_email, but someone could still reasonable see that leads to the reply_from column). 

Note some other details in the data: reply_to only appears to include the email, when reply from includes name and email both. I am not sure if this is Perceval behavior, so please let me know what you think of this.